addons/binding/org.openhab.binding.boschindego/src/main/java/org/openhab/binding/boschindego/handler/BoschIndegoHandler.java

```diff
@@ -0,0 +1,249 @@
+            DeviceStateInformation state = controller.getState();
+            DeviceStatus statusWithMessage = DeviceStatus.decodeStatusCode(state.getState());
+            int eshStatus = getEshStatusFromCommand(statusWithMessage.getAssociatedCommand());
+            int mowed = state.getMowed();
+            int error = state.getError();
+            int statecode = state.getState();
+            boolean ready = isReadyToMow(state.getState(), state.getError());
+
+            if (verifyCommand(commandToSend, eshStatus, state.getState(), error)) {
+                logger.debug("Sending command...");
+                updateState(TEXTUAL_STATE, new StringType("Refreshing..."));
+                controller.sendCommand(getCommandFromEshStatus(commandToSend));
+                commandToSend = 0;
+                try {
+                    for (int i = 0; i < 30; i++) {
+                        DeviceStateInformation stateTmp = controller.getState();
+                        if (state.getState() != stateTmp.getState()) {
+                            state = stateTmp;
+                            statusWithMessage = DeviceStatus.decodeStatusCode(state.getState());
+                            eshStatus = getEshStatusFromCommand(statusWithMessage.getAssociatedCommand());
+                            mowed = state.getMowed();
+                            error = state.getError();
+                            statecode = state.getState();
+                            ready = isReadyToMow(state.getState(), state.getError());
+                            break;
+                        }
+                        Thread.sleep(1000);
+                    }
+
+                } catch (InterruptedException e) {
+                    controller.disconnect();
+                    throw e;
+                }
+            }
+            controller.disconnect();
+            updateStatus(ThingStatus.ONLINE);
+            updateState(STATECODE, new DecimalType(statecode));
+            updateState(READY, new DecimalType(ready ? 1 : 0));
+            updateState(ERRORCODE, new DecimalType(error));
+            updateState(MOWED, new PercentType(mowed));
+            updateState(STATE, new DecimalType(eshStatus));
+            updateState(TEXTUAL_STATE, new StringType(statusWithMessage.getMessage()));
+
+        } catch (IndegoAuthenticationException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR);
+        } catch (IndegoException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+        }
+    }
+
+    private boolean isReadyToMow(int statusCode, int error) {
+        return (statusCode == 258 || statusCode == 260 || statusCode == 261 || statusCode == 517 || statusCode == 519)
+                && error == 0;
+    }
+
+    private boolean verifyCommand(int command, int eshStatus, int statusCode, int errorCode) {
+        // Mower reported an error
+        if (errorCode != 0) {
```

Comment:
if you use enums/constants here, it would be much easier to grasp what's going on here